File: core/4ueAj6/4ueAj6.py
class Node:
    def __init__(self, val=None, next=None):
        self.val = val
        self.next = next


def insert(head: 'Node', insertVal: int) -> 'Node':
    node = Node(insertVal)
    if head is None:
        node.next = node
        return node
    if head.next == head:
        head.next = node
        node.next = head
        return head

    pre_head, point = head, head.next
    while point != head:
        if pre_head.val <= insertVal <= point.val:
            break
        if pre_head.val > point.val and (insertVal < point.val or insertVal > pre_head.val):
            break
        pre_head = pre_head.next
        point = point.next

    pre_head.next = node
    node.next = point

    return head

File: core/4ueAj6/4ueAj6.java
package com.dbc.code;

public class U4eAj6 {
    class Node {
        public int val;
        public Node next;

        public Node() {}

        public Node(int _val) {
            val = _val;
        }

        public Node(int _val, Node _next) {
            val = _val;
            next = _next;
        }
    }

    public Node insert(Node head, int insertVal) {
        Node node = new Node(insertVal);
        if (head == null) {
            node.next = node;
            return node;
        }
        if (head == head.next) {
            head.next = node;
            node.next = head;
            return head;
        }
        Node preHead = head, point = head.next;
        while (point != head) {
            if (insertVal >= preHead.val && insertVal <= point.val) break;
            if (preHead.val > point.val && (insertVal > preHead.val || insertVal < point.val)) break;
            preHead = preHead.next;
            point = point.next;
        }
        preHead.next = node;
        node.next = point;
        return head;
    }
}
